src/structures/paging/mapper/mod.rs

```diff
@@ -282,7 +282,20 @@ pub trait Mapper<S: PageSize> {
     /// Removes a mapping from the page table and returns the frame that used to be mapped.
     ///
     /// Note that no page tables or pages are deallocated.
-    fn unmap(&mut self, page: Page<S>) -> Result<(PhysFrame<S>, MapperFlush<S>), UnmapError>;
+    #[inline]
+    fn unmap(&mut self, page: Page<S>) -> Result<(PhysFrame<S>, MapperFlush<S>), UnmapError> {
+        match self.unmap_ignore_present(page)? {
+            UnmappedFrame::Present { frame, flush } => Ok((frame, flush)),
+            UnmappedFrame::NotPresent { entry: _ } => Err(UnmapError::PageNotMapped),
+        }
+    }
```

Comment:
This default implementation based on `unmap_ignore_present` changes the behavior when encountering non-present pages. Previously we didn't change the entry for non-present pages, but with this default implementation, the entry will be cleared. If I saw this method returning `Err(UnmapError::PageNotMapped)` I wouldn't expect it to change the entry.

Reply:
I reverted the default implementation of unmap. You're right. This is just a bug.